pkg/asset/installconfig/azure/validation_test.go

```diff
@@ -681,10 +681,9 @@ func Test_validateResourceGroup(t *testing.T) {
 		err:       `^\Qplatform.azure.resourceGroupName: Invalid value: "valid-resource-group-conf-tags": resource group has conflicting tags kubernetes.io_cluster.test-cluster-12345\E$`,
 	}, {
 		groupName: "valid-resource-group-with-resources",
-		// ARO provisions Azure resources before resolving the asset graph,
-		// so there will always be resources in its resource group.
-		wantSkip: (&azure.Platform{}).IsARO(),
-		err:      `^\Qplatform.azure.resourceGroupName: Invalid value: "valid-resource-group-with-resources": resource group must be empty but it has 3 resources like id1, id2 ...\E$`,
+		err:       `^\Qplatform.azure.resourceGroupName: Invalid value: "valid-resource-group-with-resources": resource group must be empty but it has 3 resources like id1, id2 ...\E$`,
+	}, {
```

Comment:
Why do we expect the resource group to be empty? Maybe I don't understand this code enough but most ARO customers have many things beyond the ARO cluster in the resource group. So I believe this check would be breaking.

Reply:
Fixed it. AFAIK, if the kubernetes.io tag doesn't exist or if the resource group isn't named <infraid>-rg, the installer will not blindly delete the resource group but there's no check and delete code for the inside of the resource group to see what resources the installer created and delete just those.